### PR TITLE
Pattern improvements

### PR DESCRIPTION
**Pull request by bjoernricks**
**What**:

Rework usage of common patterns

**Why**:

Make the Plugin API clean and simple as possible.

**How**:

**Checklist**:

- [ ] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

**Comment by y0urself** [comment]
For `require_ports` and `require_udp_ports` I have added the following regex value pattern: https://regex101.com/r/3kx0jO/1
For `xref` it should be fine with: https://regex101.com/r/ynfZ0L/5

**Comment by y0urself** [comment]
We are still very slow. I ran this locally and it took nealy 40min.
```
ℹ Time elapsed: 0:37:52.903985
  Plugin                                   Error Count
  ----------------------------------------------------
× check_updated_date_version                    158740
× check_script_version_and_last_modification_tags       28154
× check_dependencies                               185
× check_includes                                  2153
× check_copyright_text                            1076
× check_script_copyright                          1076
× check_using_display                              130
× check_todo_tbd                                   142
× check_get_kb_on_services                           6
× check_grammar                                     23
× check_badwords                                    42
× check_prod_svc_detect_in_vulnvt                   17
× check_misplaced_compare_in_if                    322
× check_copyright_year                             218
× check_forking_nasl_funcs                           8
  ----------------------------------------------------
× sum                                           192292
```

I have increased the spacing between the columns because of very long check names.

File: tests/helper/test_patterns.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        patterns1 = get_script_tag_patterns()
+        patterns2 = get_script_tag_patterns()
+
+        self.assertIs(patterns1, patterns2)
+
+    def test_same_pattern_instance(self):
+        pattern1 = get_script_tag_pattern(ScriptTag.AFFECTED)
+        pattern2 = get_script_tag_pattern(ScriptTag.AFFECTED)
+
+        self.assertIs(pattern1, pattern2)
```
Comment (y0urself) [inline tests/helper/test_patterns.py:41]: The init will be done for every thread, AFAIK.

File: tests/plugins/__init__.py [hunk 2]
```diff
@@ -17,15 +17,6 @@
...
-)
-
 
 class PluginTestCase(unittest.TestCase):
-    def setUp(self) -> None:
-        self.tag_pattern = ScriptTagPatterns().pattern
-        self.special_tag_pattern = SpecialScriptTagPatterns().pattern
-
-        return super().setUp()
+    pass
```
Comment (y0urself) [inline tests/plugins/__init__.py:22]: Pretty obsolete Class now ... Remove it again?

**Review by y0urself — APPROVED** [review]
Approving your changes. Disabled auto-merge because of my changes. You can merge, if you are okay with them.

File: tests/plugins/__init__.py (hunk 2 above)
Comment (bjoernricks) [reply tests/plugins/__init__.py:22]: Yes at the moment. But I have some things in mind for making the plugin tests easier to write and therefore I kept the class.

Comment (y0urself) [reply tests/plugins/__init__.py:22]: Ok cool!

File: tests/helper/test_patterns.py (hunk 1 above)
Comment (bjoernricks) [reply tests/helper/test_patterns.py:41]: No, threads share memory. If you mean the processes from the Pool then the answer is different. When forking a process the already loaded data gets "copied" into the new process which is "copy on write" on Linux. Therefore the init is run before the forking. But at least is is only created once per process.

**Pull request merged**